pkg/scheduler/preemption/preemption_oracle.go

```diff
@@ -34,24 +35,24 @@ type PreemptionOracle struct {
 	snapshot  *cache.Snapshot
 }
 
-// IsReclaimPossible determines if a ClusterQueue can fit this
-// FlavorResource by reclaiming its nominal quota which it lent to its
-// Cohort.
-func (p *PreemptionOracle) IsReclaimPossible(log logr.Logger, cq *cache.ClusterQueueSnapshot, wl workload.Info, fr resources.FlavorResource, quantity int64) bool {
-	if cq.BorrowingWith(fr, quantity) {
-		return false
-	}
-	for _, candidate := range p.preemptor.getTargets(&preemptionCtx{
+// SimulatePreemption runs the preemption algorithm for a given flavor resource to check if
+// preemption and reclaim are possible in this flavor resource.
+func (p *PreemptionOracle) SimulatePreemption(log logr.Logger, cq *cache.ClusterQueueSnapshot, wl workload.Info, fr resources.FlavorResource, quantity int64) preemptioncommon.PreemptionPossibility {
+	candidates := p.preemptor.getTargets(&preemptionCtx{
 		log:               log,
 		preemptor:         wl,
 		preemptorCQ:       p.snapshot.ClusterQueue(wl.ClusterQueue),
 		snapshot:          p.snapshot,
 		frsNeedPreemption: sets.New(fr),
 		workloadUsage:     workload.Usage{Quota: resources.FlavorResourceQuantities{fr: quantity}},
-	}) {
+	})
+	if len(candidates) == 0 {
+		return preemptioncommon.None
+	}
+	for _, candidate := range candidates {
 		if candidate.WorkloadInfo.ClusterQueue == cq.Name {
-			return false
+			return preemptioncommon.PriorityBased
```

Comment:
Seeing the implementation I think the name `PriorityBased` is misleading. This mode may be used when priority-based preemption (within CQ) is mixed with reclamation. It only means there is at least one target which is priority-based.

I would suggest to rename it, maybe: `HasPriorityBasedTargets`, or just `Preempt`.

Reply:
fyi https://github.com/kubernetes-sigs/kueue/pull/5695